modules/calibrations/src/main/scala/lucuma/odb/calibrations/Main.scala

```diff
@@ -122,23 +126,37 @@ object CMain extends MainParams {
       sso.get(Authorization(Credentials.Token(CIString("Bearer"), c.serviceJwt)))
 
   def topics[F[_]: Concurrent: Logger](pool: Resource[F, Session[F]]):
-   Resource[F, Topic[F, ObservationTopic.Element]] =
+   Resource[F, (Topic[F, ObservationTopic.Element], Topic[F, CalibTimeTopic.Element])] =
     for {
       sup <- Supervisor[F]
       ses <- pool
+      ctt <- Resource.eval(CalibTimeTopic(ses, 1024, sup))
       top <- Resource.eval(ObservationTopic(ses, 1024, sup))
-    } yield top
+    } yield (top, ctt)
 
   def runCalibrationsDaemon[F[_]: Async: Logger](
-    obsTopic: Topic[F, ObservationTopic.Element],  services: Resource[F, Services[F]]
+    obsTopic: Topic[F, ObservationTopic.Element], calibTopic: Topic[F, CalibTimeTopic.Element], services: Resource[F, Services[F]]
   ): Resource[F, Unit] =
     for {
       _  <- Resource.eval(Logger[F].info("Start listening for program changes"))
       _  <- Resource.eval(obsTopic.subscribe(100).evalMap { elem =>
               services.useTransactionally{
                 for {
-                  t <- Sync[F].delay(Instant.now())
-                  _ <- calibrationsService.recalculateCalibrations(elem.programId, t)
+                  t <- Sync[F].delay(LocalDate.now(ZoneOffset.UTC))
+                  _ <- calibrationsService
+                        .recalculateCalibrations(
+                          elem.programId,
+                          LocalDateTime.of(t, LocalTime.MIDNIGHT).toInstant(ZoneOffset.UTC)
+                        )
+                } yield ()
+              }
+            }.compile.drain.start.void)
+      _  <- Resource.eval(Logger[F].info("Start listening for calibration time changes"))
+      _  <- Resource.eval(calibTopic.subscribe(100).evalMap { elem =>
+              services.useTransactionally{
+                for {
+                  t <- Sync[F].delay(LocalDate.now(ZoneOffset.UTC))
+                  _ <- calibrationsService.recalculateCalibrationTarget(elem.programId, elem.observationId)
```

Comment:
At first glance I thought that there wasn't a time element to this, but I believe it is taking the time from the observation time in the observation.  Maybe a comment?  It says "Start listening for calibration time changes" which. I think means the observation time of the calibration observation?

Reply:
Right it listens for events every time a the observation time of a calibration observation changes